src/customer/customer.service.ts

```diff
@@ -36,4 +39,44 @@ export class CustomerService {
     }
   }
 
+  async listAllCustomers() {
+    return this.prisma.customer.findMany() as unknown as GetCustomerDto[];
+  }
+
+  async updateAllCustomerInfo(id: string, {email, name, phone, credit, groups}: UpdatePutCustomerDto) {
+
+    if (email === undefined) {
```

Comment:
Essa implementação que vc fez de ifs não faz sentido em um método onde vc quer atualizar toda a entidade. Se vc quer atualizar todos os valores, pq ter valores que não estão sendo enviados? Se for ter um método para atualizar toda a entidade, todos os valores precisam ser enviados.